integration_tests/e2e/goal/viewArchivedGoals.cy.ts

```diff
@@ -130,4 +146,19 @@ context('Unarchive a goal', () => {
       .lastUpdatedHintAtPositionContains(1, 'Archived on 01 January 2024 by Alex Smith, Moorland (HMP & YOI)')
       .archiveReasonHintAtPositionContains(1, 'Reason: Other - Some other reason')
   })
+
+  it('should be able to navigate to the view archived goals page and have it display a message when there are no archived goals', () => {
+    // Given
+    cy.signIn()
+    cy.task('getGoalsByStatus', { status: GoalStatusValue.ARCHIVED, goals: [] })
+
+    // When
+    cy.visit(`/plan/${prisonNumber}/view/archived-goals`)
+    Page.verifyOnPage(ViewArchivedGoalsPage)
+    const archivedGoalsPage = Page.verifyOnPage(ViewArchivedGoalsPage)
+
+    // Then
+    archivedGoalsPage //
+      .noArchivedGoalsMessageContains('Daniel Craig has no archived goals')
```

Comment:
No doubt you've found some examples of where we do this type of thing, but generally we are trying not to have assertions like this that assert specific text.
The problem is that it makes the test quite fragile, because if (when! 🙄 ) the business change their mind about the wording it will break the test. EG they change it to `<prisoner name> doesn't have any goals that have been archived yet`

Some might argue that making a change to the code should break a test, and for the most part I would agree. But probably not for wording/content changes? 🤔 

I think a better approach here, and something we are trying to do elsewhere in similar tests, is to test that the element exists (pretty much the inverse of your assertion that tests it doesnt exist)
Functionally thats all we really care about - we have logic in the nunjucks template that conditionally displays the `<p>` element, so its right to test it. But I think all we need to do is test the element exists when it is supposed to, and does not exists when it is not suppposed to.

What do you think?

Reply:
This makes sense, have amended.